contracts/zero-ex/contracts/src/features/MultiplexFeature.sol

```diff
@@ -0,0 +1,799 @@
+                soldAmount = soldAmount.safeAdd(inputTokenAmount);
+                outputTokenAmount = outputTokenAmount.safeAdd(outputTokenAmount_);
+            } else if (wrappedCall.selector == this._sellToLiquidityProvider.selector) {
+                (address provider, bytes memory auxiliaryData) = abi.decode(
+                    wrappedCall.data,
+                    (address, bytes)
+                );
+                if (fillData.inputToken.isTokenETH()) {
+                    inputTokenAmount = LibSafeMathV06.min256(
+                        inputTokenAmount,
+                        remainingEth
+                    );
+                    // Transfer the input ETH to the provider.
+                    payable(provider).transfer(inputTokenAmount);
+                    // Count that ETH as spent.
+                    remainingEth = remainingEth -= inputTokenAmount;
+                } else {
+                    // Transfer input ERC20 tokens to the provider.
+                    _transferERC20Tokens(
+                        fillData.inputToken,
+                        msg.sender,
+                        provider,
+                        inputTokenAmount
+                    );
+                }
+                // Perform the PLP trade.
+                uint256 outputTokenAmount_ = _sellToLiquidityProvider(
+                    fillData.inputToken,
+                    fillData.outputToken,
+                    inputTokenAmount,
+                    ILiquidityProvider(provider),
+                    msg.sender,
+                    auxiliaryData
+                );
+                // Increment the sold and bought amounts.
+                soldAmount = soldAmount.safeAdd(inputTokenAmount);
+                outputTokenAmount = outputTokenAmount.safeAdd(outputTokenAmount_);
+            } else if (wrappedCall.selector == ITransformERC20Feature._transformERC20.selector) {
+                ITransformERC20Feature.TransformERC20Args memory args;
+                args.taker = msg.sender;
+                args.inputToken = fillData.inputToken;
+                args.outputToken = fillData.outputToken;
+                args.inputTokenAmount = inputTokenAmount;
+                args.minOutputTokenAmount = 0;
+                uint256 ethValue;
+                (args.transformations, ethValue) = abi.decode(
+                    wrappedCall.data,
+                    (ITransformERC20Feature.Transformation[], uint256)
+                );
+                // Normalize the ETH and input token amounts.
+                ethValue = _normalizeAmount(
+                    ethValue,
+                    totalEth,
+                    remainingEth
+                );
+                try ITransformERC20Feature(address(this))._transformERC20
+                    {value: ethValue}
+                    (args)
+                    returns (uint256 outputTokenAmount_)
+                {
+                    remainingEth -= ethValue;
```

Comment:
Would we tell FQT to transfer unspent protocol fees back to us or the taker?

Reply:
back to the taker